src/lib/features/change-request-access-service/sql-change-request-access-read-model.ts

```diff
@@ -0,0 +1,52 @@
+import { SKIP_CHANGE_REQUEST } from '../../types';
+import { Db } from '../../db/db';
+import { AccessService } from '../../services';
+import User from '../../types/user';
+import { IChangeRequestAccessReadModel } from './change-request-access-read-model';
+
+export class ChangeRequestAccessReadModel
+    implements IChangeRequestAccessReadModel
+{
+    private db: Db;
+
+    private accessService: AccessService;
+
+    constructor(db: Db, accessService: AccessService) {
+        this.db = db;
+        this.accessService = accessService;
+    }
+
+    public async canBypassChangeRequest(
+        project: string,
+        environment: string,
+        user?: User,
+    ): Promise<boolean> {
+        const [canSkipChangeRequest, changeRequestEnabled] = await Promise.all([
+            user
+                ? this.accessService.hasPermission(
+                      user,
+                      SKIP_CHANGE_REQUEST,
+                      project,
+                      environment,
+                  )
+                : Promise.resolve(false),
+            this.accessService.isChangeRequestsEnabled(project, environment),
```

Comment:
From your comment above maybe you meant this instead? Otherwise I don't see the point of the `async isChangeRequestsEnabled` method in the class.

```suggestion
            this.isChangeRequestsEnabled(project, environment),
```

Reply:
Brilliant finding @nunogois Thanks!